contracts/AutonomousDripper.sol

```diff
@@ -0,0 +1,187 @@
+        address[] memory _assetsHeld = new address[](assetsWatchlist.length);
+        uint256 count = 0;
+        for (uint idx = 0; idx < assetsWatchlist.length; idx++) {
+            uint256 balance = IERC20(assetsWatchlist[idx]).balanceOf(address(this));
+            if (balance > 0) {
+                _assetsHeld[count] = assetsWatchlist[idx];
+                count++;
+            }
+        }
+        if (count != assetsWatchlist.length) {
+            assembly {
+                mstore(_assetsHeld, count)
+            }
+        }
+        return _assetsHeld;
+    }
+
+    /**
+     * @dev Runs off-chain at every block to determine if the `performUpkeep`
+     * function should be called on-chain.
+     */
+    function checkUpkeep(bytes calldata) external view override whenNotPaused returns (
+        bool upkeepNeeded, bytes memory checkData
+    ) {
+        if ((block.timestamp - lastTimestamp) > interval) {
+            address[] memory assetsHeld = _getAssetsHeld();
+            if (assetsHeld.length > 0) {
+                return (true, abi.encode(assetsHeld));
+            }
+        }
+    }
+
+    /**
+     * @dev Contains the logic that should be executed on-chain when
+     * `checkUpkeep` returns true.
+     */
+    function performUpkeep(bytes calldata performData) external override onlyKeeperRegistry whenNotPaused {
+        if ((block.timestamp - lastTimestamp) > interval) {
+            address[] memory assetsHeld = abi.decode(performData, (address[]));
+            for (uint idx = 0; idx < assetsHeld.length; idx++) {
+                if (IERC20(assetsHeld[idx]).balanceOf(address(this)) > 0) {
+                    VestingWallet.release(assetsHeld[idx]);
+                    lastTimestamp = block.timestamp;
+                }
+            }
+        }
+    }
+
+    /**
+     * @dev Sweep the full contract's ether balance to the current owner. Can
+     * only be called by the current owner.
+     */
+    function sweep() public virtual onlyOwner {
+        uint256 balance = address(this).balance;
+        emit EtherSwept(balance);
+        Address.sendValue(payable(super.owner()), balance);
+    }
+
+    /**
+     * @dev Sweep the full contract's balance for an ERC-20 token to the
+     * current owner. Can only be called by the current owner.
+     */
+    function sweep(address token) public virtual onlyOwner {
+        uint256 balance = IERC20(token).balanceOf(address(this));
+        emit ERC20Swept(token, balance);
+        SafeERC20.safeTransfer(IERC20(token), super.owner(), balance);
+    }
+
+    /**
+    * @dev Getter for the keeper registry address.
+    */
+    function getKeeperRegistryAddress() external view returns (address keeperRegistryAddress) {
+        return _keeperRegistryAddress;
+    }
+
+    /**
+    * @dev Setter for the keeper registry address.
+    */
+    function setKeeperRegistryAddress(address keeperRegistryAddress) public onlyOwner {
+        require(keeperRegistryAddress != address(0));
```

Comment:
Can change this to a custom error for consistency

Reply:
yes you are right.

however i opted for removing it all together (7b117b7), since there is no security risk to setting it to 0 and can always be reset.